presidio-anonymizer/tests/test_analyzer_result.py

```diff
@@ -0,0 +1,235 @@
+    [
+        ("bla", 0.2, 4, 10),
+        ("changed", 0.2, 0, 10),
+        ("bla", 0.2, 0, 11),
+        ("bla", 0.3, 0, 10),
+    ],
+    # fmt: on
+)
+def test_analyzer_result_not_equal_another(entity_type, score, start, end):
+    first = create_analayzer_result("bla", 0.2, 0, 10)
+    second = create_analayzer_result(entity_type, score, start, end)
+
+    assert first != second
+
+
+def test_analyzer_result_successfully_hashed_and_equal():
+    first = create_analayzer_result("", 0, 0, 10)
+    second = create_analayzer_result("", 0, 0, 10)
+
+    assert first.__hash__() == second.__hash__()
+
+
+@pytest.mark.parametrize(
+    # fmt: off
+    "entity_type, score, start, end",
+    [
+        ("bla", 0.2, 4, 10),
+        ("changed", 0.2, 0, 10),
+        ("bla", 0.2, 0, 11),
+        ("bla", 0.3, 0, 10),
+    ],
+    # fmt: on
+)
+def test_analyzer_result_hash_not_equal_another(entity_type, score, start, end):
+    first = create_analayzer_result("bla", 0.2, 0, 10)
+    second = create_analayzer_result(entity_type, score, start, end)
+
+    assert first.__hash__() != second.__hash__()
+
+
+@pytest.mark.parametrize(
+    # fmt: off
+    "entity_type, score, start, end",
+    [
+        ("bla", 0.2, 0, 10),
+        ("changed", 0.2, 2, 10),
+        ("bla", 0.3, 0, 11),
+        ("bla", 0.1, 0, 10),
+    ],
+    # fmt: on
+)
+def test_analyzer_result_has_conflict(entity_type, score, start, end):
+    first = create_analayzer_result("bla", 0.2, 2, 10)
+    second = create_analayzer_result(entity_type, score, start, end)
+
+    assert first.has_conflict(second)
+
+
+@pytest.mark.parametrize(
+    # fmt: off
+    "entity_type, score, start, end",
+    [
+        ("bla", 0.2, 3, 10),
+        ("changed", 0.1, 2, 10),
+        ("bla", 0.3, 0, 9),
+    ],
+    # fmt: on
+)
+def test_analyzer_result_has_no_conflict(entity_type, score, start, end):
+    first = create_analayzer_result("bla", 0.2, 2, 10)
+    second = create_analayzer_result(entity_type, score, start, end)
+
+    assert not first.has_conflict(second)
+
+
+@pytest.mark.parametrize(
+    # fmt: off
+    "request_json, result_text",
+    [
+        ({}, "Invalid input, analyzer result must contain start",),
+        ({
+             "end": 32,
+             "score": 0.8,
+             "entity_type": "NUMBER"
+         }, "Invalid input, analyzer result must contain start",),
+        ({
+             "start": 28,
+             "score": 0.8,
+             "entity_type": "NUMBER"
+         }, "Invalid input, analyzer result must contain end",),
+        ({
+             "start": 28,
+             "end": 32,
+             "entity_type": "NUMBER"
+         }, "Invalid input, analyzer result must contain score",),
+        ({
+             "start": 28,
+             "end": 32,
+             "score": 0.8,
+         }, "Invalid input, analyzer result must contain entity_type",),
+    ],
+    # fmt: on
+)
+def test_analyzer_result_fails_on_invalid_json_formats(request_json, result_text):
```

Comment:
pytest has support for raised exceptions https://docs.pytest.org/en/stable/assert.html#assertions-about-expected-exceptions.

Reply:
What would be the output if I'm missing more than one input field?

Reply:
Will change to the supported raised exceptions.